vexflow/vexflow.d.ts

```diff
@@ -0,0 +1,1325 @@
+            setContext(context : IRenderContext) : StaveHairpin;
+            setPosition(position : Modifier.Position) : StaveHairpin;
+            setRenderOptions(options : {height? : number, y_shift : number, left_shift_px : number, right_shift_px : number}) : StaveHairpin;
+            setNotes(notes : {first_note : Note, last_note : Note}) : StaveHairpin;
+            draw() : boolean;
+        }
+        
+        export module StaveLine {
+            enum TextVerticalPosition {TOP, BOTTOM}
+            enum TextJustification {LEFT, CENTER, RIGHT}
+        }
+        
+        class StaveLine {
+            constructor(notes : {first_note: Note, last_note: Note, first_indices : number[], last_indices : number[]});
+            setContext(context : Object) : StaveLine;
+            setFont(font : {family : string, size : number, weight : string}) : StaveLine;
+            setText(text : string) : StaveLine;
+            setNotes(notes : {first_note: Note, last_note: Note, first_indices? : number[], last_indices? : number[]}) : StaveLine;
+            applyLineStyle() : void;
+            applyFontStyle() : void;
+            draw() : StaveLine;
+            
+            //inconsistent API: this should be set via an options object in the constructor
+            render_options : {padding_left : number, padding_right : number, line_width : number, line_dash : number[], rounded_end : boolean, color : string, draw_start_arrow : boolean, draw_end_arrow : boolean, arrowhead_length : number, arrowhead_angle : number, text_position_vertical : StaveLine.TextVerticalPosition, text_justification : StaveLine.TextJustification};
+        }
+        
+        class StaveModifier {
+            getCategory() : string;
+            makeSpacer(padding : number) : {getContext: Function, setStave: Function, renderToStave: Function, getMetrics: Function};
+            placeGlyphOnLine(glyph : Glyph, stave : Stave, line : number) : void;
+            setPadding(padding : number) : void;
+            addToStave(stave : Stave, firstGlyph : boolean) : StaveModifier;
+            addToStaveEnd(stave : Stave, firstGlyph : boolean) : StaveModifier;
+            addModifier() : void;
+            addEndModifier() : void;
+        }
+        
+        class StaveNote extends StemmableNote {
+            //TODO remove the following lines once TypeScript allows subclass overrides with type changes and/or inconsistencies mentioned below are fixed
+            buildStem() : StemmableNote;            
+            setStave(stave : Stave) : Note;
+            addModifier(modifier : Modifier, index? : number) : Note;
+            getModifierStartXY() : {x : number, y : number};
+            getDots() : number;
+
+            constructor(note_struct : {type : string, dots? : number, duration : string, clef : string, keys : string[], octave_shift? : number});
```

Comment:
On StaveNote class you have:
```
constructor(note_struct : {type : string, dots? : number, duration : string, clef : string, keys : string[], octave_shift? : number});
```

The examples I see do not set  `type` so that StaveNote is used as: `new Vex.Flow.StaveNote({ keys: ["c/4"], duration: "q" }),`. Was `type` meant to be optional in this constructor (it is optional in the super class `StemmableNote`).